src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,131 @@
+
+            context.RegisterSyntaxNodeAction(ctx =>
+            {
+                ctx.CancellationToken.ThrowIfCancellationRequested();
+
+                // Checks whether the declared method is an orchestration, if not, returns
+                if (ctx.ContainingSymbol is not IMethodSymbol methodSymbol ||
+                    !methodSymbol.ContainsAttributeInAnyMethodArguments(knownSymbols.OrchestrationTriggerAttribute) ||
+                    !methodSymbol.TryGetSingleValueFromAttribute(knownSymbols.FunctionAttribute, out string functionName))
+                {
+                    return;
+                }
+
+                var orchestration = new OrchestrationMethod(functionName, methodSymbol);
+                var methodSyntax = (MethodDeclarationSyntax)ctx.Node;
+
+                FindInvokedMethods(ctx.SemanticModel, methodSyntax, methodSymbol, orchestration, result);
+            }, SyntaxKind.MethodDeclaration);
+
+            // allows concrete implementations to register specific actions/analysis and then check if they happen in any of the orchestration methods
+            this.RegisterAdditionalCompilationStartAction(context, result);
+        });
+    }
+
+    // Recursively find all methods invoked by the orchestration method
+    static void FindInvokedMethods(
+        SemanticModel semanticModel, MethodDeclarationSyntax callerSyntax, IMethodSymbol callerSymbol,
+        OrchestrationMethod rootOrchestration, OrchestrationAnalysisResult result)
+    {
+        if (!TryTrackMethod(semanticModel, callerSyntax, callerSymbol, rootOrchestration, result))
+        {
+            // previously tracked method, leave to avoid infinite recursion
+            return;
+        }
+
+        foreach (InvocationExpressionSyntax invocationSyntax in callerSyntax.DescendantNodes().OfType<InvocationExpressionSyntax>())
+        {
+            IOperation? operation = semanticModel.GetOperation(invocationSyntax);
+            if (operation == null || operation is not IInvocationOperation invocation)
+            {
+                continue;
+            }
+
+            IMethodSymbol calleeMethodSymbol = invocation.TargetMethod;
+            if (calleeMethodSymbol == null)
+            {
+                continue;
+            }
+
+            // iterating over multiple syntax references is needed because the same method can be declared in multiple places (e.g. partial classes)
+            IEnumerable<MethodDeclarationSyntax> calleeSyntaxes = calleeMethodSymbol.DeclaringSyntaxReferences.Select(r => r.GetSyntax()).OfType<MethodDeclarationSyntax>();
+            foreach (MethodDeclarationSyntax calleeSyntax in calleeSyntaxes)
+            {
+                FindInvokedMethods(semanticModel, calleeSyntax, calleeMethodSymbol, rootOrchestration, result);
+            }
+        }
+    }
+
+    static bool TryTrackMethod(SemanticModel semanticModel, MethodDeclarationSyntax callerSyntax, IMethodSymbol callerSymbol,
```

Comment:
Can you add a comment explaining what this method does?

Reply:
In the most recent commit, I cleaned up and removed the method. 
The code is the same, just moved to the caller location. Essentially, it adds the visited method to the dictionary.